lightning/src/routing/router.rs

```diff
@@ -1520,12 +1530,89 @@ where L::Target: Logger {
 	Ok(route)
 }
 
+// Add a random 'shadow route' offset to the CLTV expiry delta values to improve privacy.
```

Comment:
nit: "Only the *final* hop is offset".

Reply:
Also nice if you can add a quick note explaining which privacy is improved here.

E.g "to prevent intermediate node to observe the payment destination". Or refer to the BOLT 7 recommendation.

Reply:
Thanks, I now tried to include a more descriptive comment regarding the issue!